assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useLoadMedia from './actions/useLoadMedia';
+import useReloadMedia from './actions/useReloadMedia';
+import Context from './context';
+
+function MediaProvider( { children } ) {
+	const [ media, setMedia ] = useState( [] );
+	const [ mediaType, setMediaType ] = useState( '' );
+	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ isMediaLoaded, setIsMediaLoaded ] = useState( false );
+	const [ isMediaLoading, setIsMediaLoading ] = useState( false );
+
+	const loadMedia = useLoadMedia( { setMedia, setIsMediaLoading, setIsMediaLoaded, isMediaLoaded, isMediaLoading, mediaType, searchTerm } );
+	const reloadMedia = useReloadMedia( { setIsMediaLoading, setIsMediaLoaded } );
+
+	const state = {
+		state: {
+			media,
+			isMediaLoading,
+			isMediaLoaded,
+			mediaType,
+			searchTerm,
```

Comment:
Does each element type / tab track their own search term or is the search term shared between the tabs?

Reply:
Search term is passed the API but is also needs by the library. It is a shame here to put this in global scope,  but there is no way around it.